fe/fe-core/src/main/java/com/starrocks/task/AgentBatchTask.java

```diff
@@ -121,11 +116,12 @@ public boolean isFinished() {
     // return the limit number of unfinished tasks.
     public List<AgentTask> getUnfinishedTasks(int limit) {
         List<AgentTask> res = Lists.newArrayList();
-        for (List<AgentTask> tasks : this.backendIdToTasks.values()) {
-            for (AgentTask agentTask : tasks) {
+        for (Map.Entry<Long, List<AgentTask>> entry : this.backendIdToTasks.entrySet()) {
+            for (AgentTask agentTask : entry.getValue()) {
                 if (!agentTask.isFinished()) {
-                    if (res.size() < limit) {
-                        res.add(agentTask);
+                    res.add(agentTask);
+                    if (res.size() >= limit) {
+                        return res;
```

Comment:
This modification changes the original behaviour.
Whereas the previous behaviour would collect all matches, this modification returns the first non-conforming one it encounters.

Reply:
I don't quite understand,  IMHO, both implementations will push the first `limit` unfinished tasks into `res`, the difference is that the previous one will iterate all elements, even if the size of `res` has reached to `limit`.

Reply:
Oh, you're right.
I misunderstand it.